src/main/java/io/elastic/jdbc/ProcedureFieldsNameProvider.java

```diff
@@ -0,0 +1,137 @@
+        });
+
+    JsonObjectBuilder inMetadata = Json.createObjectBuilder()
+        .add("type", "object")
+        .add("properties", inFields.build());
+
+    parameters.stream()
+        .filter(p -> p.getDirection() == Direction.OUT || p.getDirection() == Direction.INOUT)
+        .forEach(p -> {
+          JsonObjectBuilder valueContent = Json.createObjectBuilder()
+              .add("type", Utils.cleanJsonType(Utils.detectColumnType(p.getType(), "")))
+              .add("name", p.getName())
+              .add("required", true);
+          outFields.add(p.getName(), valueContent.build());
+        });
+
+    JsonObjectBuilder outMetadata = Json.createObjectBuilder()
+        .add("type", "object")
+        .add("properties", outFields.build());
+
+    result.add("in", inMetadata.build()).add("out", outMetadata.build());
+
+    JsonObject metadataResponse = result.build();
+    System.out.println(metadataResponse);
+    return metadataResponse;
+  }
+
+  public List<String> getProceduresList(JsonObject config) {
+    List<String> result = new ArrayList<>();
+    try (Connection conn = Utils.getConnection(config)) {
+      DatabaseMetaData meta = conn.getMetaData();
+      ResultSet res = meta.getProcedures(null, config.getString("schemaName"), null);
+      System.out.println("List of procedures: ");
+      while (res.next()) {
+        String cat = res.getString("PROCEDURE_CAT");
+        String schem = res.getString("PROCEDURE_SCHEM");
+        String name = res.getString("PROCEDURE_NAME");
+
+        result.add(name);
+      }
+      res.close();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    return result;
+  }
+
+  public static List<ProcedureParameter> getProcedureMetadata(JsonObject config) {
+    List<ProcedureParameter> parameters = new LinkedList<>();
+
+    try (Connection conn = Utils.getConnection(config)) {
+      Statement st = conn.createStatement();
+
+      DatabaseMetaData dbMetaData = conn.getMetaData();
+      ResultSet rs = dbMetaData.getProcedureColumns(conn.getCatalog(),
+          config.getString("schemaName"),
+          config.getString("procedureName"),
+          null);
+
+      while (rs.next()) {
+        // get stored procedure metadata
+        String procedureCatalog = rs.getString(1);
+        String procedureSchema = rs.getString(2);
+        String procedureName = rs.getString(3);
+        String columnName = rs.getString(4);
+        short columnReturn = rs.getShort(5);
+        int columnDataType = rs.getInt(6);
+        String columnReturnTypeName = rs.getString(7);
+        int columnPrecision = rs.getInt(8);
+        int columnByteLength = rs.getInt(9);
+        short columnScale = rs.getShort(10);
+        short columnRadix = rs.getShort(11);
+        short columnNullable = rs.getShort(12);
+        String columnRemarks = rs.getString(13);
+
+        parameters.add(new ProcedureParameter(columnName, columnReturn, columnDataType));
+      }
+
+      st.close();
```

Comment:
What if there will be an exception thrown somewhere above?
Will statement be closed in this case?